test/analysis/headerCheck/src/main.cpp

```diff
@@ -20,7 +20,16 @@
  */
 
 #define BOOST_TEST_MODULE headerCheck
+
+#include <boost/predef.h>   // BOOST_COMP_CLANG
+#if BOOST_COMP_CLANG
+    #pragma clang diagnostic push
+    #pragma clang diagnostic ignored "-Wunused-parameter"
+#endif
 #include <boost/test/unit_test.hpp>
+#if BOOST_COMP_CLANG
+    #pragma clang diagnostic pop
```

Comment:
Is there a need to disable all this pragma's. 
I know that unknown pragmas result in a warning but maybe we should disable the warning for unknown pragmas.